src/wasm/wasm-type.cpp

```diff
@@ -125,7 +133,19 @@ Type::Type(std::initializer_list<Type> types) { init(types); }
 
 Type::Type(const std::vector<Type>& types) { init(types); }
 
-size_t Type::size() const { return expand().size(); }
+size_t Type::size() {
+  if (_size == unknownSize) {
+    _size = expand().size();
+  }
+  return _size;
+}
+
+size_t Type::size() const {
+  if (_size == unknownSize) {
+    return expand().size();
```

Comment:
this is the bad old path? i hope we can avoid it as mentioned in another comment, by requiring the size to always be known.

Reply:
After discussion, this seems ok for now but we should figure this out later.